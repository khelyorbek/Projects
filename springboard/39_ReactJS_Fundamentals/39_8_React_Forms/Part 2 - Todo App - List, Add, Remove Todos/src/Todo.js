import React from 'react';

// this component should display a div with a background color, width and height based on the props passed to it.
const Todo = (props) => {
    const handleRemove = (e) => {
        props.removeTodo(props.todoTitle);
    }

    return (<li>{props.todoTitle} <button onClick={handleRemove}>X</button></li>)
}

export default Todo;