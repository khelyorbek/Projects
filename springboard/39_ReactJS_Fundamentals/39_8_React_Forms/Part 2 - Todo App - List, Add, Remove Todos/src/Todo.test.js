import React from 'react';
import { render } from "@testing-library/react";
import Todo from './Todo';

// smoke test
it('renders without crashing', function() {
  render(<Todo />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<Todo />);
  expect(asFragment()).toMatchSnapshot();
});

