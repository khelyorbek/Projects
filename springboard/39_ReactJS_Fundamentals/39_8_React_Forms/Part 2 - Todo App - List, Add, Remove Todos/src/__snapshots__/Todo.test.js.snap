// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li>
     
    <button>
      X
    </button>
  </li>
</DocumentFragment>
`;
